layer/csv_logging.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STADIA_OPEN_SOURCE_PERFORMANCE_LAYERS_CSV_LOGGING_H_
+#define STADIA_OPEN_SOURCE_PERFORMANCE_LAYERS_CSV_LOGGING_H_
+
+#include <cstdio>
+#include <string>
+
+#include "event_logging.h"
+#include "layer_data.h"
+
+namespace performancelayers {
+// Takes an `Event` instance as an input and generates a csv string containing
+// `event`'s name and attribute values.
+// TODO(miladhakimi): Differentiate hashes and other integers. Hashes
+// should be displayed in hex.
+std::string EventToCSVString(Event &event);
+
+// CSVLogger logs the events in the CSV format to the output given in its
+// constructor. Sample use:
+// ```c++
+// CSVLogger logger("pipeline,duration", "compile_time.csv");
+// logger.StartLog();
+// Event compile_time_event = ...;
+// logger.AddEvent(&compile_time_event);
+// logger.Flush();
+// logger.EndLog();
+// ```
+// There is no need to add '\n' at the end of the csv_header in the constructor.
+// This is handled by the implementation.
+class CSVLogger : public EventLogger {
+ public:
+  CSVLogger(const char *csv_header, const char *filename)
+      : header_(csv_header) {
+    if (filename) {
+      out_ = fopen(filename, "w");
```

Comment:
What happens when `fopen` fails? E.g., end used provides a bad path

Reply:
go/cstyle#Doing_Work_in_Constructors talks briefly about error handling in constructors. There are a couple of options to deal with `fopen` failure:
1. Using stderr instead.
2. exiting the program.
3. Take the file descriptor as an argument. In this case, the user is responsible for handling the `fopen` failure.
4. Set an `is_valid` flag and have a public getter for it. The user should check if the logger is valid before using the methods.

Reply:
Continuing with stderr sounds fine to me. This way the user can know the layer is working but may be misconfigured

Reply:
We could also print an error message to stderr when the filename was not provided or was bad

Reply:
(Using SPL_LOG)